src/uct/ib/rc/verbs/rc_verbs_ep.c

```diff
@@ -575,14 +618,19 @@ ucs_status_t uct_rc_verbs_ep_connect_to_ep(uct_ep_h tl_ep,
                                         &path_mtu);
     ucs_assert(path_mtu != UCT_IB_ADDRESS_INVALID_PATH_MTU);
 
-    qp_num = uct_ib_unpack_uint24(rc_addr->qp_num);
+    qp_num = uct_ib_unpack_uint24(rc_addr->super.qp_num);
     status = uct_rc_iface_qp_connect(iface, ep->qp, qp_num, &ah_attr, path_mtu);
     if (status != UCS_OK) {
         return status;
     }
 
-    if (rc_addr->flags & UCT_RC_VERBS_ADDR_HAS_ATOMIC_MR) {
-        ep->super.atomic_mr_offset = uct_ib_md_atomic_offset(*(uint8_t*)(rc_addr + 1));
+    if (rc_addr->super.flags & UCT_RC_VERBS_ADDR_HAS_ATOMIC_MR) {
+        ep->super.atomic_mr_offset = uct_ib_md_atomic_offset(rc_addr->atomic_mr_id);
+
+        if (rc_addr->super.flags & UCT_RC_VERBS_ADDR_FLUSH_REMOTE) {
```

Comment:
seems that with verbs, atomic_mr and flush-remote always come together
so need to remove UCT_RC_VERBS_ADDR_FLUSH_REMOTE flag and rely on UCT_RC_VERBS_ADDR_HAS_ATOMIC_MR

Reply:
> seems that with verbs, atomic_mr and flush-remote always come together
> so need to remove UCT_RC_VERBS_ADDR_FLUSH_REMOTE flag and rely on UCT_RC_VERBS_ADDR_HAS_ATOMIC_MR

done